`agent/common/issuerservice` tests fail to compile

When running `go test ./agent/common/issuerservice` the test run fails with the following output:

```
agent/common/issuerservice/issuer_test.go:60:85: not enough arguments in call to client.CreateHolder
        have (string, string, string)
        want (context.Context, string, string, string)
agent/common/issuerservice/issuer_test.go:75:85: not enough arguments in call to client.CreateHolder
        have (string, string, string)
        want (context.Context, string, string, string)
agent/common/issuerservice/issuer_test.go:101:85: not enough arguments in call to client.CreateHolder
        have (string, string, string)
        want (context.Context, string, string, string)
agent/common/issuerservice/issuer_test.go:126:61: not enough arguments in call to client.RevokeCredential
        have (string, string)
        want (context.Context, string, string)
agent/common/issuerservice/issuer_test.go:151:61: not enough arguments in call to client.RevokeCredential
        have (string, string)
        want (context.Context, string, string)
agent/common/issuerservice/issuer_test.go:176:61: not enough arguments in call to client.RevokeCredential
        have (string, string)
        want (context.Context, string, string)
agent/common/issuerservice/issuer_test.go:201:29: not enough arguments in call to client.DeleteHolder
        have (string)
        want (context.Context, string)
agent/common/issuerservice/issuer_test.go:226:29: not enough arguments in call to client.DeleteHolder
        have (string)
        want (context.Context, string)
agent/common/issuerservice/issuer_test.go:250:29: not enough arguments in call to client.DeleteHolder
        have (string)
        want (context.Context, string)
FAIL    github.com/eclipse-cfm/cfm/agent/common/issuerservice [build failed]
FAIL
```

The tests in `agent/common/issuerservice/issuer_test.go` are stale and still call the client methods without the required `context.Context` parameter.

 Suggested fix: update the test calls to pass a context, e.g. `context.Background()`.

--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,7 @@ EDCV_DIR=agent/edcv
 KEYCLOAK_DIR=agent/keycloak
 REG_DIR=agent/registration
 ONBOARDING_DIR=agent/onboarding
+AGENT_COMMON=agent/common
 KIND_CLUSTER_NAME=edcv
 
 E2E_DIR=e2e
@@ -101,6 +102,7 @@ test: install-gotestsum
 	$(MAKE) -C $(REG_DIR) test
 	$(MAKE) -C $(ONBOARDING_DIR) test
 	$(MAKE) -C $(ASSEMBLY_DIR) test
+	$(MAKE) -C $(AGENT_COMMON) test
 
 test-common:
 	@echo "Testing common..."
@@ -125,6 +127,10 @@ test-reg:
 test-onboarding:
 	@echo "Testing Onboarding agent..."
 	$(MAKE) -C $(ONBOARDING_DIR) test
+
+test-agent-common:
+	@echo "Testing agent common..."
+	$(MAKE) -C $(AGENT_COMMON) test
 
 #==============================================================================
 # Development Commands - Delegate to Service Makefiles

--- a/agent/common/Makefile
+++ b/agent/common/Makefile
@@ -1,0 +1,17 @@
+cd.PHONY: test clean
+
+# Build settings
+BUILD_DIR=bin
+
+# Environment variables
+export CGO_ENABLED=0
+
+
+# Run tests
+test:
+	$(TEST_CMD)
+
+# Clean build artifacts
+clean:
+	rm -rf $(BUILD_DIR)
+	go clean

--- a/agent/common/identityhub/identityhub_test.go
+++ b/agent/common/identityhub/identityhub_test.go
@@ -24,6 +24,7 @@ import (
 
 	"github.com/eclipse-cfm/cfm/agent/edcv"
 	"github.com/eclipse-cfm/cfm/common/mocks"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
@@ -72,7 +73,7 @@ func TestIdentityAPIClient_CreateParticipantContext(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("token", nil)
+	tp.On("GetToken", mock.Anything).Return("token", nil)
 	client := HttpIdentityAPIClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -88,14 +89,14 @@ func TestIdentityAPIClient_CreateParticipantContext(t *testing.T) {
 			manifest.VaultCredentials.ClientID = "client-id"
 			manifest.VaultCredentials.TokenURL = "https://example.com/idp/token"
 		})
-	_, err := client.CreateParticipantContext(manifest)
+	_, err := client.CreateParticipantContext(t.Context(), manifest)
 	require.NoError(t, err)
 
 }
 
 func TestIdentityAPIClient_AuthError(t *testing.T) {
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("", fmt.Errorf("test error"))
+	tp.On("GetToken", mock.Anything).Return("", fmt.Errorf("test error"))
 	client := HttpIdentityAPIClient{
 		BaseURL:       "http://foo.bar",
 		TokenProvider: tp,
@@ -111,7 +112,7 @@ func TestIdentityAPIClient_AuthError(t *testing.T) {
 			manifest.VaultCredentials.ClientID = "client-id"
 			manifest.VaultCredentials.TokenURL = "https://example.com/idp/token"
 		})
-	_, err := client.CreateParticipantContext(manifest)
+	_, err := client.CreateParticipantContext(t.Context(), manifest)
 	require.ErrorContains(t, err, "failed to get API access token: test error")
 }
 
@@ -123,7 +124,7 @@ func TestIdentityAPIClient_BadRequest(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("token", nil)
+	tp.On("GetToken", mock.Anything).Return("token", nil)
 	client := HttpIdentityAPIClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -139,7 +140,7 @@ func TestIdentityAPIClient_BadRequest(t *testing.T) {
 			manifest.VaultCredentials.ClientID = "client-id"
 			manifest.VaultCredentials.TokenURL = "https://example.com/idp/token"
 		})
-	_, err := client.CreateParticipantContext(manifest)
+	_, err := client.CreateParticipantContext(t.Context(), manifest)
 	require.ErrorContains(t, err, "foobar")
 }
 
@@ -156,14 +157,14 @@ func TestHttpIdentityAPIClient_DeleteParticipantContext(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("token", nil)
+	tp.On("GetToken", mock.Anything).Return("token", nil)
 	client := HttpIdentityAPIClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.DeleteParticipantContext("test-id")
+	err := client.DeleteParticipantContext(t.Context(), "test-id")
 	require.NoError(t, err)
 }
 
@@ -175,13 +176,13 @@ func TestHttpIdentityAPIClient_DeleteParticipantContext_NotFound(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("token", nil)
+	tp.On("GetToken", mock.Anything).Return("token", nil)
 	client := HttpIdentityAPIClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.DeleteParticipantContext("test")
-	require.ErrorContains(t, err, "participant not found")
+	err := client.DeleteParticipantContext(t.Context(), "test")
+	require.ErrorContains(t, err, "participant context test not found")
 }

--- a/agent/common/issuerservice/issuer_test.go
+++ b/agent/common/issuerservice/issuer_test.go
@@ -24,6 +24,7 @@ import (
 	"testing"
 
 	"github.com/eclipse-cfm/cfm/common/mocks"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
@@ -49,7 +50,7 @@ func TestHttpApiClient_CreateHolder(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -57,14 +58,14 @@ func TestHttpApiClient_CreateHolder(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.CreateHolder("did:web:test-participant", "did:web:test-participant", "test holder")
+	err := client.CreateHolder(t.Context(), "did:web:test-participant", "did:web:test-participant", "test holder")
 	require.NoError(t, err)
 }
 
 func TestHttpApiClient_CreateHolder_AuthError(t *testing.T) {
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("", fmt.Errorf("test error"))
+	tp.On("GetToken", mock.Anything).Return("", fmt.Errorf("test error"))
 	client := HttpApiClient{
 		BaseURL:       "http://foo.bar",
 		TokenProvider: tp,
@@ -72,7 +73,7 @@ func TestHttpApiClient_CreateHolder_AuthError(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.CreateHolder("did:web:test-participant", "did:web:test-participant", "test holder")
+	err := client.CreateHolder(t.Context(), "did:web:test-participant", "did:web:test-participant", "test holder")
 	require.ErrorContains(t, err, "test error")
 }
 
@@ -90,7 +91,7 @@ func TestHttpApiClient_CreateHolder_ApiReturnsError(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -98,7 +99,7 @@ func TestHttpApiClient_CreateHolder_ApiReturnsError(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.CreateHolder("did:web:test-participant", "did:web:test-participant", "test holder")
+	err := client.CreateHolder(t.Context(), "did:web:test-participant", "did:web:test-participant", "test holder")
 	require.ErrorContains(t, err, "failed to create Holder")
 }
 
@@ -115,7 +116,7 @@ func TestHttpApiClient_RevokeCredential(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -123,7 +124,7 @@ func TestHttpApiClient_RevokeCredential(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.RevokeCredential("did:web:test-participant", "test-credential-id")
+	err := client.RevokeCredential(t.Context(), "did:web:test-participant", "test-credential-id")
 	require.NoError(t, err)
 }
 
@@ -140,7 +141,7 @@ func TestHttpApiClient_RevokeCredential_ClientError(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -148,7 +149,7 @@ func TestHttpApiClient_RevokeCredential_ClientError(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.RevokeCredential("did:web:test-participant", "test-credential-id")
+	err := client.RevokeCredential(t.Context(), "did:web:test-participant", "test-credential-id")
 	require.ErrorContains(t, err, "failed to revoke credential")
 }
 
@@ -165,7 +166,7 @@ func TestHttpApiClient_RevokeCredential_NotFound(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -173,7 +174,7 @@ func TestHttpApiClient_RevokeCredential_NotFound(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.RevokeCredential("did:web:test-participant", "test-credential-id")
+	err := client.RevokeCredential(t.Context(), "did:web:test-participant", "test-credential-id")
 	require.ErrorContains(t, err, "failed to revoke credential")
 }
 
@@ -190,7 +191,7 @@ func TestHttpApiClient_DeleteHolder(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -198,7 +199,7 @@ func TestHttpApiClient_DeleteHolder(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.DeleteHolder("did:web:test-participant")
+	err := client.DeleteHolder(t.Context(), "did:web:test-participant")
 	require.NoError(t, err)
 }
 
@@ -215,7 +216,7 @@ func TestHttpApiClient_DeleteHolder_NotFound(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -223,7 +224,7 @@ func TestHttpApiClient_DeleteHolder_NotFound(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.DeleteHolder("did:web:test-participant")
+	err := client.DeleteHolder(t.Context(), "did:web:test-participant")
 	require.ErrorContains(t, err, "received status code 404")
 }
 func TestHttpApiClient_DeleteHolder_AuthError(t *testing.T) {
@@ -239,7 +240,7 @@ func TestHttpApiClient_DeleteHolder_AuthError(t *testing.T) {
 	defer server.Close()
 
 	tp := mocks.NewMockTokenProvider(t)
-	tp.On("GetToken").Return("test token", nil)
+	tp.On("GetToken", mock.Anything).Return("test token", nil)
 	client := HttpApiClient{
 		BaseURL:       server.URL,
 		TokenProvider: tp,
@@ -247,6 +248,6 @@ func TestHttpApiClient_DeleteHolder_AuthError(t *testing.T) {
 		HttpClient:    &http.Client{},
 	}
 
-	err := client.DeleteHolder("did:web:test-participant")
+	err := client.DeleteHolder(t.Context(), "did:web:test-participant")
 	require.ErrorContains(t, err, "received status code 401")
 }

--- a/e2e/e2efixtures/client.go
+++ b/e2e/e2efixtures/client.go
@@ -189,7 +189,6 @@ func (c *ApiClient) deleteRequest(url string) error {
 	return nil
 }
 
-
 func (c *ApiClient) getRequest(baseUrl string, endpoint string, result any) error {
 	url := fmt.Sprintf("%s/%s", baseUrl, endpoint)
 	req, err := http.NewRequest("GET", url, nil)
@@ -214,5 +213,3 @@ func (c *ApiClient) getRequest(baseUrl string, endpoint string, result any) erro
 
 	return json.Unmarshal(body, result)
 }
-
-

